File: S1/python/fact.py

def factorial_loop(n):
    factorial = 1
    for i in range(1, n + 1):
        factorial *= i
    return factorial


num = int(input("Enter a number: "))
result = factorial_loop(num)
print("Factorial of", num, "is", result)

def factorial_recursive(n):
    if n == 0:
        return 1
    else:
        return n * factorial_recursive(n - 1)

num = int(input("Enter a number: "))
result = factorial_recursive(num)
print("Factorial of", num, "is", result)

File: S1/ADS/fact.c
#include<stdio.h>
void main()
{
int n,i,fact=1;
printf("Enter number");
scanf("%d",&n);
for(i=n;i>0;i--)
{
fact=fact*i;
}
printf("Factorial of %d is %d \N",n,fact);
}
